### PR TITLE
restore page configuration

### PR DESCRIPTION
**Pull request by paddeo**

### CODE REVIEW

**Comment by entando-service** [comment]
All is well

File: engine/src/main/java/org/entando/entando/aps/system/services/page/PageService.java [hunk 1]
```diff
@@ -269,6 +272,31 @@ public PageConfigurationDto getPageConfiguration(String pageCode, String status)
...
+                throw new RestRourceNotFoundException(ERRCODE_PAGE_NOT_FOUND, "page", pageCode);
+            }
+            IPage pageO = this.loadPage(pageCode, STATUS_ONLINE);
+            if (null == pageO) {
+                DataBinder binder = new DataBinder(pageCode);
+                BindingResult bindingResult = binder.getBindingResult();
+                bindingResult.reject(ERRCODE_STATUS_INVALID, new String[]{pageCode}, "page.status.invalid");
+                throw new ValidationGenericException(bindingResult);
+            }
+            ((Page) pageD).setMetadata(pageO.getMetadata());
```
Comment (joewhite101) [inline engine/src/main/java/org/entando/entando/aps/system/services/page/PageService.java]: Why is this cast necessary? Should we just update the interface?

File: engine/src/main/java/org/entando/entando/web/page/PageConfigurationController.java [hunk 2]
```diff
@@ -91,14 +90,22 @@ public void setPageService(IPageService pageService) {
...
         this.getPageService().deleteWidgetConfiguration(pageCode, frameId);
         Map<String, String> metadata = new HashMap<>();
         metadata.put("status", IPageService.STATUS_DRAFT);
         return new ResponseEntity<>(new RestResponse(frameId, null, metadata), HttpStatus.OK);
     }
 
+    @RestAccessControl(permission = Permission.SUPERUSER)
+    @RequestMapping(value = "/pages/{pageCode}/configuration/restore", method = RequestMethod.PUT, produces = MediaType.APPLICATION_JSON_VALUE)
+    public ResponseEntity<?> updatePageConfiguration(@PathVariable String pageCode) {
+        logger.trace("requested {} configuration", pageCode);
```
Comment (joewhite101) [inline engine/src/main/java/org/entando/entando/web/page/PageConfigurationController.java]: Change to DEBUG

File: engine/src/main/java/org/entando/entando/web/page/PageConfigurationController.java [hunk 3]
```diff
@@ -91,14 +90,22 @@ public void setPageService(IPageService pageService) {
     @RestAccessControl(permission = Permission.SUPERUSER)
     @RequestMapping(value = "/pages/{pageCode}/widgets/{frameId}", method = RequestMethod.DELETE, produces = MediaType.APPLICATION_JSON_VALUE)
     public ResponseEntity<?> deletePageWidget(
-                                              @PathVariable String pageCode,
-                                              @PathVariable int frameId) {
+            @PathVariable String pageCode,
+            @PathVariable int frameId) {
         logger.trace("removing widget configuration in page {} and frame {}", pageCode, frameId);
```
Comment (joewhite101) [inline engine/src/main/java/org/entando/entando/web/page/PageConfigurationController.java]: Change to DEBUG

File: engine/src/main/java/org/entando/entando/web/page/PageConfigurationController.java [hunk 4]
```diff
@@ -73,14 +73,13 @@ public void setPageService(IPageService pageService) {
...
     public ResponseEntity<?> updatePageWidget(
-                                              @PathVariable String pageCode, 
-                                              @PathVariable int frameId, 
-                                              @Valid @RequestBody WidgetConfigurationRequest widget,
-                                              BindingResult bindingResult) {
+            @PathVariable String pageCode,
+            @PathVariable int frameId,
+            @Valid @RequestBody WidgetConfigurationRequest widget,
+            BindingResult bindingResult) {
         logger.trace("updating widget configuration in page {} and frame {}", pageCode, frameId);
```
Comment (joewhite101) [inline engine/src/main/java/org/entando/entando/web/page/PageConfigurationController.java]: Change to DEBUG

File: engine/src/main/java/org/entando/entando/aps/system/services/page/PageService.java [hunk 5]
```diff
@@ -269,6 +272,31 @@ public PageConfigurationDto getPageConfiguration(String pageCode, String status)
         return pageConfigurationDto;
     }
 
+    @Override
+    public PageConfigurationDto restorePageConfiguration(String pageCode) {
```
Comment (joewhite101) [inline engine/src/main/java/org/entando/entando/aps/system/services/page/PageService.java]: Add debug logging

File: engine/src/main/java/org/entando/entando/aps/system/services/page/PageService.java (hunk 1 above)
Comment (paddeo) [reply engine/src/main/java/org/entando/entando/aps/system/services/page/PageService.java]: That interface does not have setters (only getters).

Comment (joewhite101) [reply engine/src/main/java/org/entando/entando/aps/system/services/page/PageService.java]: Let's update the interface to expose these methods. Casting to the concrete implementation here isn't good.

**Comment by entando-service** [comment]
All is well

**Comment by entando-service** [comment]
All is well

**Pull request merged**